src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -898,7 +876,15 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
                 return (NotificationSeverity.Error, EditorFeaturesResources.Rename_operation_was_cancelled_or_is_not_valid);
 
             if (!_workspace.TryApplyChanges(finalSolution))
-                return (NotificationSeverity.Error, EditorFeaturesResources.Rename_operation_could_not_complete_due_to_external_change_to_workspace);
+            {
+                // If the workspace changed in TryOnBeforeGlobalSymbolRenamed retry, this prevents rename from failing for cases
+                // where text changes to other files or workspace state change doesn't impact the text changes being applied. 
+                Logger.Log(FunctionId.Rename_TryApplyRename_WorkspaceChanged, message: null, LogLevel.Information);
+                finalSolution = CalculateFinalSolutionSynchronously(newSolution, _workspace, changedDocumentIDs, cancellationToken);
+
+                if (!_workspace.TryApplyChanges(finalSolution))
+                    return (NotificationSeverity.Error, EditorFeaturesResources.Rename_operation_could_not_complete_due_to_external_change_to_workspace);
+            }
```

Comment:
💭 Can we not check `_workspace.CurrentSolution` explicitly to avoid having two calls to `TryApplyChanges`?

Reply:
From my understanding this is preferred, and TryApplyChanges will throw for all cases except workspace versions being different